src/libraries/System.IO.Compression.ZipFile/tests/ZipFile.Unix.cs

```diff
@@ -136,6 +139,42 @@ public void UnixExtractFilePermissionsCompat(string zipName, string expectedPerm
             }
         }
 
+        [Fact]
+        [PlatformSpecific(TestPlatforms.AnyUnix & ~TestPlatforms.Browser)]
+        public async Task CanZipNamedPipe()
+        {
+            string destPath = Path.Combine(TestDirectory, "dest.zip");
+
+            string subFolderPath = Path.Combine(TestDirectory, "subfolder");
+            string fifoPath = Path.Combine(subFolderPath, "namedPipe");
+            Directory.CreateDirectory(subFolderPath); // mandatory before calling mkfifo
+            Assert.Equal(0, mkfifo(fifoPath, 438 /* 666 in octal */));
+
+            byte[] contentBytes = { 1, 2, 3, 4, 5 };
+
+            await Task.WhenAll(
+                Task.Run(() =>
+                {
+                    using FileStream fs = new (fifoPath, FileMode.Open, FileAccess.Write, FileShare.Read);
+                    foreach (byte content in contentBytes)
+                    {
+                        fs.WriteByte(content);
```

Comment:
You're writing the bytes out individually... do you also want to flush?

Reply:
> do you also want to flush?

The `Dispose` method is going to do that for me:

https://github.com/dotnet/runtime/blob/ab9ee763e500649ac42ca5058cc576b6d7f96357/src/libraries/System.Private.CoreLib/src/System/IO/Strategies/BufferedFileStreamStrategy.cs#L142-L150

Reply:
I know.  But then why use Whitewater in a loop rather than just calling Write on the whole array.  Seems the only reason that'd be useful is if you actually wanted to push each byte across the pipe.

Reply:
You are right, I've forgot to disable buffering ;)